src/api/app/controllers/webui/request_controller.rb

```diff
@@ -12,11 +12,30 @@ class Webui::RequestController < Webui::WebuiController
 
   def create
     @bs_request = BsRequest.new(bs_request_params)
+    @bs_request.set_add_revision if params.key?(:add_revision)
     authorize @bs_request, :create?
 
     begin
       @bs_request.save!
     # FIXME: Use validations in the model instead of raising whenever something is wrong
+    rescue BsRequestAction::MissingAction
+      flash[:error] = 'Unable to submit, sources are unchanged'
+
+      if params.key?(:package_name)
```

Comment:
This condition is repeated three times. I think the `begin` and `rescues` could be like:

```ruby
...
    begin
      @bs_request.save!
      redirect_to request_show_path(@bs_request.number)
      return
    # FIXME: Use validations in the model instead of raising whenever something is wrong
    rescue BsRequestAction::MissingAction
      flash[:error] = 'Unable to submit, sources are unchanged'
    rescue Project::Errors::UnknownObjectError
      flash[:error] = "Unable to submit: The source of package #{params[:project_name]}/#{params[:package_name]} is broken"
    rescue APIError, ActiveRecord::RecordInvalid => e
      flash[:error] = e.message
    end

    if params.key?(:package_name)
      redirect_to(package_show_path(params[:project_name], params[:package_name]))
    else
      redirect_to(project_show_path(params[:project_name]))
    end
  end
```
```